src/main/java/org/apache/commons/imaging/formats/png/PngImageParser.java

```diff
@@ -222,58 +261,39 @@ public void readSignature(final InputStream is) throws ImageReadException,
 
     }
 
-    private List<PngChunk> readChunks(final ByteSource byteSource, final ChunkType[] chunkTypes,
-            final boolean returnAfterFirst) throws ImageReadException, IOException {
-        try (InputStream is = byteSource.getInputStream()) {
-            readSignature(is);
-            return readChunks(is, chunkTypes, returnAfterFirst);
-        }
-    }
-
     @Override
     public byte[] getICCProfileBytes(final ByteSource byteSource, final Map<String, Object> params)
             throws ImageReadException, IOException {
-        final List<PngChunk> chunks = readChunks(byteSource, new ChunkType[] { ChunkType.iCCP },
-                true);
-
-        if ((chunks == null) || (chunks.isEmpty())) {
-            // throw new ImageReadException("Png: No chunks");
+        final PngChunkIccp pngChunkiCCP = (PngChunkIccp) ICCP_SEL.filterOnlyOne(
+                ChunkReader.newInstance(byteSource).readChunks(ICCP_SEL, true));
+        
+        if(pngChunkiCCP == null) {
+            System.err.println("Apache Commons Imaging: PNG file doesn't contain ICC profile");
```

Comment:
We are now using a logging framework instead of `System.err`, and no need to include the project name. The project name changed from Sanselan to Imaging recently-ish. That could happen again, and there is no need to include it really.